# Notebook 01 - Data Understanding




## 1. Import Libraries


In [1]:
from pathlib import Path

import pandas as pd
import numpy as np

## 2. Configure Project Paths

In [2]:
# Project paths
PROJECT_ROOT = Path("..")

DATA_DIR = PROJECT_ROOT / "data"
RAW_DIR = DATA_DIR / "raw"

TRAIN_DIR = RAW_DIR / "train"
TEST_DIR = RAW_DIR / "test"

## 3.Validate Dataset Structure

In [3]:
print(f"Project Root : {PROJECT_ROOT.resolve()}")
print(f"Train Exists : {TRAIN_DIR.exists()}")
print(f"Test Exists  : {TEST_DIR.exists()}")

Project Root : C:\Dev\Rogii Wellbore Geology\rogii-wellbore-geology
Train Exists : True
Test Exists  : True


## 4. Count Training Files

In [4]:
typewell_files = sorted(TRAIN_DIR.glob("*_typewell.csv"))
horizontal_files = sorted(TRAIN_DIR.glob("*_horizontal_well.csv"))

print(f"Typewell files   : {len(typewell_files)}")
print(f"Horizontal files : {len(horizontal_files)}")

Typewell files   : 773
Horizontal files : 773


## 5. Validate Well Pairs

In [5]:
typewell_ids = {file.name.replace("_typewell.csv", "") for file in typewell_files}
horizontal_ids = {file.name.replace("_horizontal_well.csv", "") for file in horizontal_files}

complete_pairs = typewell_ids & horizontal_ids
missing_horizontal = typewell_ids - horizontal_ids
missing_typewell = horizontal_ids - typewell_ids

print(f"Complete pairs     : {len(complete_pairs)}")
print(f"Missing horizontal : {len(missing_horizontal)}")
print(f"Missing typewell   : {len(missing_typewell)}")

Complete pairs     : 773
Missing horizontal : 0
Missing typewell   : 0


## 6. Load a Sample Well Pair

In [6]:
sample_id = sorted(complete_pairs)[0]

sample_typewell_path = TRAIN_DIR / f"{sample_id}_typewell.csv"
sample_horizontal_path = TRAIN_DIR / f"{sample_id}_horizontal_well.csv"

typewell_df = pd.read_csv(sample_typewell_path)
horizontal_df = pd.read_csv(sample_horizontal_path)

print("Sample ID:", sample_id)
print("Typewell shape:", typewell_df.shape)
print("Horizontal shape:", horizontal_df.shape)

Sample ID: 000d7d20_
Typewell shape: (1296, 3)
Horizontal shape: (5278, 13)


## 7. Inspect Columns

In [7]:
print("Typewell columns:")
print(typewell_df.columns.tolist())

print("\nHorizontal well columns:")
print(horizontal_df.columns.tolist())

Typewell columns:
['TVT', 'GR', 'Geology']

Horizontal well columns:
['MD', 'X', 'Y', 'Z', 'ANCC', 'ASTNU', 'ASTNL', 'EGFDU', 'EGFDL', 'BUDA', 'TVT', 'GR', 'TVT_input']


## 8. Preview the Data

In [8]:
print("Typewell Dataset")
display(typewell_df.head())

print("\nHorizontal Well Dataset")
display(horizontal_df.head())

Typewell Dataset


,TVT,GR,Geology
0,11223.95,126.11,NaN
1,11224.45,128.22,NaN
2,11224.95,128.72,NaN
3,11225.45,128.12,NaN
4,11225.95,125.29,NaN



Horizontal Well Dataset


,MD,X,Y,Z,ANCC,ASTNU,ASTNL,EGFDU,EGFDL,BUDA,TVT,GR,TVT_input
0,11467.0,2983525.16,1069022.09,-9258.57,-9395.81,-9569.86,-9597.64,-9670.99,-9705.96,-9846.35,11236.02,115.692586,11236.02
1,11468.0,2983525.18,1069022.30,-9259.55,-9395.75,-9569.80,-9597.58,-9670.93,-9705.90,-9846.29,11237.05,115.584293,11237.05
2,11469.0,2983525.20,1069022.52,-9260.52,-9395.69,-9569.74,-9597.52,-9670.87,-9705.84,-9846.23,11238.09,135.446960,11238.09
3,11470.0,2983525.22,1069022.73,-9261.50,-9395.64,-9569.69,-9597.47,-9670.82,-9705.79,-9846.18,11239.12,140.401346,11239.12
4,11471.0,2983525.25,1069022.95,-9262.47,-9395.58,-9569.63,-9597.41,-9670.76,-9705.73,-9846.12,11240.15,111.270638,11240.15


## 9. Dataset Information

In [9]:
print("Typewell Information")
typewell_df.info()

print("\n----------------------------------------\n")

print("Horizontal Well Information")
horizontal_df.info()

Typewell Information
<class 'pandas.DataFrame'>
RangeIndex: 1296 entries, 0 to 1295
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   TVT      1296 non-null   float64
 1   GR       1296 non-null   float64
 2   Geology  997 non-null    str    
dtypes: float64(2), str(1)
memory usage: 30.5 KB

----------------------------------------

Horizontal Well Information
<class 'pandas.DataFrame'>
RangeIndex: 5278 entries, 0 to 5277
Data columns (total 13 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   MD         5278 non-null   float64
 1   X          5278 non-null   float64
 2   Y          5278 non-null   float64
 3   Z          5278 non-null   float64
 4   ANCC       5278 non-null   float64
 5   ASTNU      5278 non-null   float64
 6   ASTNL      5278 non-null   float64
 7   EGFDU      5278 non-null   float64
 8   EGFDL      5278 non-null   float64
 9   BUDA       5278 non-null   float6

## 10. Statistical Summary

In [10]:
print("Typewell Summary")
display(typewell_df.describe())

print("\nHorizontal Well Summary")
display(horizontal_df.describe())

Typewell Summary


,TVT,GR
count,1296.000000,1296.000000
mean,11547.700000,83.257639
std,187.133642,26.251321
min,11223.950000,28.660000
25%,11385.825000,64.940000
50%,11547.700000,87.090000
75%,11709.575000,102.802500
max,11871.450000,158.180000



Horizontal Well Summary


,MD,X,Y,Z,ANCC,ASTNU,ASTNL,EGFDU,EGFDL,BUDA,TVT,GR,TVT_input
count,5278.000000,5.278000e+03,5.278000e+03,5278.000000,5278.000000,5278.000000,5278.000000,5278.000000,5278.000000,5278.000000,5278.000000,3020.000000,1442.000000
mean,14105.500000,2.983514e+06,1.071418e+06,-9673.194324,-9330.621821,-9504.671821,-9532.451821,-9605.801821,-9640.771821,-9781.161821,11715.831127,94.436961,11646.417864
std,1523.771691,2.997379e+01,1.503381e+03,80.961903,35.272977,35.272977,35.272977,35.272977,35.272977,35.272977,88.782092,18.627493,148.869554
min,11467.000000,2.983468e+06,1.069022e+06,-9755.610000,-9395.810000,-9569.860000,-9597.640000,-9670.990000,-9705.960000,-9846.350000,11236.020000,31.765827,11236.020000
25%,12786.250000,2.983487e+06,1.070090e+06,-9721.762500,-9362.307500,-9536.357500,-9564.137500,-9637.487500,-9672.457500,-9812.847500,11735.850000,85.068343,11566.195000
50%,14105.500000,2.983514e+06,1.071407e+06,-9689.230000,-9335.975000,-9510.025000,-9537.805000,-9611.155000,-9646.125000,-9786.515000,11742.130000,94.936506,11745.920000
75%,15424.750000,2.983535e+06,1.072723e+06,-9650.035000,-9296.665000,-9470.715000,-9498.495000,-9571.845000,-9606.815000,-9747.205000,11747.257500,103.960888,11747.465000
max,16744.000000,2.983578e+06,1.074041e+06,-9258.570000,-9271.000000,-9445.050000,-9472.830000,-9546.180000,-9581.150000,-9721.540000,11756.120000,217.352257,11756.120000


## 11. Check Missing Values


In [11]:
print("Typewell Missing Values")
display(typewell_df.isnull().sum())

print("\nHorizontal Missing Values")
display(horizontal_df.isnull().sum())

Typewell Missing Values


TVT          0
GR           0
Geology    299
dtype: int64


Horizontal Missing Values


MD              0
X               0
Y               0
Z               0
ANCC            0
ASTNU           0
ASTNL           0
EGFDU           0
EGFDL           0
BUDA            0
TVT             0
GR           2258
TVT_input    3836
dtype: int64

## 12. Initial Observations

- Successfully loaded a complete training well pair.
- The training dataset contains 773 typewell files and 773 horizontal well files, with no missing pairs.
- The typewell dataset contains geological labels (`Geology`) together with `TVT` and `GR` measurements.
- The horizontal well dataset contains trajectory information, formation markers, and well log measurements.
- Missing values are present in several columns, particularly `Geology` in the typewell and `GR` and `TVT_input` in the horizontal well.
- `TVT` appears in both datasets and is likely to be an important feature for relating the vertical and horizontal wells.
- Additional exploratory analysis is required to understand the relationship between the two wells before feature engineering and model development.